data_structures/src/transaction_factory.rs

```diff
@@ -5,57 +5,70 @@ use crate::{
     },
     error::TransactionError,
     transaction::{DRTransactionBody, VTTransactionBody},
-    utxo_pool::{OwnUnspentOutputsPool, UnspentOutputsPool, UtxoDiff, UtxoSelectionStrategy},
+    utxo_pool::{
+        NodeUtxos, OwnUnspentOutputsPool, UnspentOutputsPool, UtxoDiff, UtxoSelectionStrategy,
+    },
 };
 use std::{collections::HashSet, convert::TryFrom};
 
+/// Abstraction used to create transactions using a UtxoSet
+pub trait UtxoFantasy {
```

Comment:
```suggestion
/// Abstraction that facilitates creation of new transactions from a set of unspent outputs.
/// Transaction factories are expected to operate on this trait so that their business logic can be applied on many heterogeneous data structures that may implement it. 
pub trait OutputsCollection {
```